core/chain-configs/src/genesis_config.rs

```diff
@@ -337,6 +457,33 @@ impl Genesis {
         hasher.process_genesis(self);
         hasher.finalize()
     }
+
+    pub fn stream_records_with_callback(&self, callback: impl FnMut(StateRecord)) {
```

Comment:
Same here, I'd return an `io::Result` here  (you can convert `serde_json::Error` to `io::Error` https://docs.rs/serde_json/1.0.64/src/serde_json/error.rs.html#129-168)